src/main/java/com/microsoft/jenkins/kubernetes/wrapper/ResourceManager.java

```diff
@@ -84,23 +82,13 @@ final String getKind() {
         /**
          * Explicitly apply the configuration if a resource with the same name exists in the namespace in the cluster,
          * or create one if not.
-         * <p>
-         * If we cannot load resource during application (possibly because the resource gets deleted after we first
-         * checked), or some one created the resource after we checked and before we created, the method fails with
-         * exception.
-         *
-         * @throws IOException if we cannot find the resource in the cluster when we apply the configuration
          */
-        final void createOrApply() throws IOException {
+        final void createOrApply() {
             T original = getCurrentResource();
             T current = get();
             T updated;
             if (original != null) {
                 updated = applyResource(original, current);
-                if (updated == null) {
```

Comment:
No more need nullPointer check?

Reply:
If an `ApiException` occurs, a `RuntimeException` is thrown to interrupt the pipeline, so no `null `value is returned.